lib/linguist/heuristics.yml

```
@@ -484,6 +484,13 @@ disambiguations:
   - language: NL
     pattern: '^(b|g)[0-9]+ '
   - language: NewLisp
+- extensions: ['.nr']
+  rules:
+  - language: Noir
+    and:
+    - negative_pattern: '^\.'
+    - pattern: '^\s*fn .*\(.*'
```

```suggestion
    - pattern: '^\s*fn[ \t]+[^\(\s].*?\('
```

This accounts for multiple whitespace characters after an `fn` keyword, and also requires at least one non-whitespace, non-bracket character to be matched before the opening parenthesis.

---

ah thanks for flagging. I'll commit and update the search query link in the original comment